processing/src/main/java/org/apache/druid/query/groupby/having/HavingSpecMetricComparator.java

```diff
@@ -73,6 +73,9 @@ static int compare(String aggregationName, Number value, Map<String, AggregatorF
           double d = Double.parseDouble(metricValueStr);
           return Double.compare(d, value.doubleValue());
         }
+      } else if (metricValueObj instanceof Number) {
+        BigDecimal d = new BigDecimal(metricValueObj.toString());
```

Comment:
This is going to convert metricValueObj to a string and then parse it as a big decimal in every row, which seems pretty wasteful.